R/MsBackendMzR-functions.R

```diff
@@ -116,3 +116,183 @@ MsBackendMzR <- function() {
     }
     res[, columns, drop = FALSE]
 }
+
+#' Function to write the MS data (for a single file/sample) to an mzML/mzXML
+#' file.
+#'
+#' @param x `Spectra` object with the data
+#'
+#' @param file `character(1)` with the (full) output file name.
+#'
+#' @param format `character(1)` with the format.
+#'
+#' @param verbose `logical(1)` verbosity.
+#'
+#' @param copy `logical(1)` whether general file information should be copied
+#'     from the original MS file. Note this only works on `Spectra` with a
+#'     `MsBackendMzR` backend as it assumes `dataStorage` contains the original
+#'     MS data file.
+#'
+#' @noRd
+#'
+#' @author Johannes Rainer
+.write_ms_data_mzR <- function(x, file, format = c("mzML", "mzXML"),
+                               software_processing = NULL,
+                               verbose = TRUE,
+                               copy = FALSE) {
+    match.arg(tolower(format), c("mzml", "mzxml"))
+    if (verbose)
+        message("Writing file ", basename(file), "...", appendLF = FALSE)
+    file <- path.expand(file)
+    if (copy && length(unique(dataOrigin(x))) == 1) {
+        original_file <- unique(dataOrigin(x))
```

Comment:
Because you already test `length(unique(dataOrigin(x))) == 1` the following lines could be collapsed to:
`if (!file.exists(as.character(dataOrigin(x)[1])) { warning(...); copy <- FALSE }`
(would save an `if`, an `any` and an `unique` statement)

Reply:
Excellent! Thanks. I will change.

Reply:
Done in https://github.com/rformassspectrometry/Spectra/commit/b725a7e96605b58fe6102308181b3f6fa7c7f706